libraries/chain/include/eosio/chain/contracts/contract_table_objects.hpp

```diff
@@ -46,7 +46,18 @@ namespace eosio { namespace chain { namespace contracts {
    struct by_scope_primary;
    struct by_scope_secondary;
    struct by_scope_tertiary;
-
+   struct by_bta;
+
+   template <typename ObjectType>
+   using bta_index = ordered_unique<tag<by_bta>,
+      composite_key<ObjectType,
+                    member<ObjectType, typename ObjectType::id_type, &ObjectType::id>,
+                    member<ObjectType, table_id, &ObjectType::t_id>,
+                    member<ObjectType, account_name, &ObjectType::bta>
```

Comment:
can we demote the `ObjectType::id_type` member of this index to the last member of the composite?

That would allow for queries in the form of "all rows in a given table that I am paying for".   I think the format as present means you have to have id values in the query

Reply:
Done.